src/swap/saga.ts

```diff
@@ -340,6 +341,12 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
     }
 
     yield* put(swapSuccess({ swapId, fromTokenId, toTokenId }))
+    yield* put(
+      trackPointsEvent({
```

Comment:
thanks for calling this out!

i was thinking about it too, and opted for an imperative dispatch in place rather than a subscription elsewhere.

i agree about the debugger visibility as a benefit of having a separate dispatch.

also, it looked easy to understand to group all the post-swap events in one place. that way we can easily see what should happen once the swap succeeds:
* set appropriate state status
* dispatch points event
* send analytics event